x/concentrated-liquidity/range_test.go

```diff
@@ -439,7 +439,7 @@ func (s *KeeperTestSuite) getInitialPositionAssets(pool types.ConcentratedPoolEx
 
 	// Calculate asset amounts that would be required to get the required spot price (rounding up on asset1 to ensure we stay in the intended tick)
 	asset0Amount := sdk.NewInt(100000000000000)
-	asset1Amount := sdk.NewDecFromInt(asset0Amount).Mul(requiredPrice).Ceil().TruncateInt()
+	asset1Amount := sdk.NewDecFromInt(asset0Amount).Mul(requiredPrice.SDKDec()).Ceil().TruncateInt()
```

Comment:
Asking to learn: for these places where we turn bigdec into sdk dec, how do we ensure it is safe to do truncations?

Reply:
Our [underlying sqrt function](https://github.com/osmosis-labs/osmosis/blob/99349ff5e8c6d3d59fc02ffe5640c0788b4b1302/x/concentrated-liquidity/math/tick.go#L45) operates on the `sdk.Dec`.

As a result, all tick sqrt prices have 18 decimals despite being `osmomath.BigDec` with precision of 36.

That is, they get computed to something like 1.123456789123456789000000000000000000` where the last 18 decimals are just zeroes.

However, our "current sqrt price" can fluctuate beyond 18th decimal.

This PR makes the API for sqrt prices to be consistently 36 decimals. However, this change is state-compatible because we simply truncate tick data from 36 to 18 decimals in [different places](https://github.com/osmosis-labs/osmosis/blob/99349ff5e8c6d3d59fc02ffe5640c0788b4b1302/x/concentrated-liquidity/math/tick.go#L42) while ensuring 18 decimals by [the sqrt function](url) despite having a [36 decimal API](https://github.com/osmosis-labs/osmosis/blob/99349ff5e8c6d3d59fc02ffe5640c0788b4b1302/x/concentrated-liquidity/math/tick.go#L34).

In the future, this change will allow us to have two strategies for computing sqrt prices from ticks and vice versa. One with 18 decimals, and the other one with 36. As a result, we will be able to lower our min spot price for the strategy with 36 decimals. 18 decimals will continue to exist as is, thus, not requiring a state migration